adam/curriculum_to_html.py

```diff
@@ -0,0 +1,184 @@
+        a *title* for the pages the generator loops through each group and calls the internal 
+        method to create HTML pages. *overwrite* indicates is previously existing HTML files should
+        be overwritten or not
+        """
+        for x in range(len(instances)):  # pylint:disable=consider-using-enumerate
+            CurriculumToHtml._generate(
+                instance=instances[x],
+                outputdestination=f"{outputdestination}{title}{x}.html",
+                title=title + f" {x}",
+                overwrite=overwrite,
+            )
+
+    @staticmethod
+    def _generate(
+        instance: InstanceGroup[
+            HighLevelSemanticsSituation,
+            LinearizedDependencyTree,
+            DevelopmentalPrimitivePerceptionFrame,
+        ],
+        title: str,
+        overwrite: bool,
+        outputdestination: str,
+    ):
+        """
+        Internal generation method for individual instance groups into HTML pages
+
+        Given an `InstanceGroup` with a `HighLevelSemanticsSituation`,
+        `LinearizedDependencyTree`, and `DevelopmentalPrimitivePerceptionFrame` this function
+        creates an html page at the given *outputdestination* and *title*. If the file already
+        exists and *overwrite* is set to False an error is raised in execution. Each page turns an
+        instance group with each "instance" as an indiviudal section on the page.
+
+        No returns
+
+        """
+        if os.path.isfile(outputdestination) and not overwrite:
+            raise RuntimeError(f"Not able to create new HTML file in {outputdestination}")
+        html = open(outputdestination, "w")
+        html.write(f"<h1>{title} - {instance.name()}</h1>\n")
+        instance_number = 0
+        for inst in instance.instances():
+            if not isinstance(inst[0], HighLevelSemanticsSituation):
+                raise RuntimeError(
+                    f"Expected the Situation to be HighLevelSemanticsSituation got {type(inst[0])}"
+                )
+            if not isinstance(inst[1], LinearizedDependencyTree):
+                raise RuntimeError(
+                    f"Expected the Lingustics to be LinearizedDependencyTree got {type(inst[1])}"
+                )
+            if not isinstance(inst[2], PerceptualRepresentation) and not isinstance(
+                inst[2].frames, DevelopmentalPrimitivePerceptionFrame
+            ):
+                raise RuntimeError(
+                    f"Expected the Perceptual Representation to contain DevelopmentalPrimitivePerceptionFrame got {type(inst[2].frames)}"
+                )
+            html.write(
+                f'<table>\n<thead><tr><th colspan="3"><h2>Scene {instance_number}</h2>\n</th></tr></thead>'
+                f'<tbody><tr><td><h3 id="situation{instance_number}">Situation Description</h3></td>'
+                f'<td><h3 id="lingustics{instance_number}">Linguistic Descrption</h3></td>'
```

Comment:
separating the instance number from `linguistics` by a dash will make the HTML more readable